staging/src/k8s.io/apiserver/pkg/registry/generic/rest/streamer.go

```diff
@@ -61,11 +64,28 @@ func (s *LocationStreamer) InputStream(ctx context.Context, apiVersion, acceptHe
 	if transport == nil {
 		transport = http.DefaultTransport
 	}
+	requestAddress := s.Location.Host
+	klog.Infof("Sending request to %q.", requestAddress)
+
+	if httpTransport, ok := transport.(*http.Transport); ok {
```

Comment:
our debugging round trippers often wrap http.Transport... they all implement RoundTripperWrapper which lets you drill down to the innermost one

Reply:
or better yet, have the thing constructing the locationstreamer build a transport with the correct dialer in the first place... modifying the guts of a wrapped thing is always a bit sketchy (the current uses of RoundTripperWrapper are all about *extracting* information, not making modifications

Reply:
Done